### PR TITLE
add 'exo add help' command

### PR DESCRIPTION
**Pull request by hugobho**
@kevgo Related to https://github.com/Originate/exosphere-sdk/pull/158

### CODE REVIEW

File: features/help.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Feature: help command
+
+  As an Exosphere user who isn't familiar with the available 'exo' commands
```
Comment (kevgo) [inline features/help.feature]: As a developer not familiar with how to call "exo add" correctly

File: features/help.feature [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  So that I can do the things I intended without having to look this up somewhere else.
+
+
+  Scenario: the user enters 'exo add help'
+    When running "exo-add help" in the terminal
+    Then I see:
+    """
+    Usage: exo add [options]
+
+    options: [<service-name>] [<template>] [<model>] [<description>]
```
Comment (kevgo) [inline features/help.feature]: I think the first parameter is the entity name, then comes the service name?

Comment (kevgo) [inline features/help.feature]: Lets format this like git help is formatted:

```
usage: exo-add [<entity-name>]...

Adds a service to this application....

File: features/help.feature [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        Adds a reusable Exoservice next to application directory to be used by multiple applications.
```
Comment (kevgo) [inline features/help.feature]: I think exo-create creates the reusable service, and exo-add adds a service to the current application.

File: features/steps/steps.ls [hunk 4]
```diff
@@ -37,6 +37,19 @@ module.exports = ->
       cb!
     async.each table.rows!, enter-input, done
 
+  @When /^running "([^"]*)" in the terminal$/, timeout: 30_000, (command, done) ->
```
Comment (kevgo) [inline features/steps/steps.ls]: Why the 30 sec timeout here?

File: src/cli.ls [hunk 5]
```diff
@@ -42,6 +46,16 @@ inquirer.prompt(questions).then (answers) ->
 function service-names
   fs.readdir-sync path.join(templates-path, 'add-service')
 
+function help
+  console.log "\nUsage: " + cyan 'exo' + blue ' add' + yellow " [options]\n"
+  help-message =
+    """
+    options: #{yellow '[<service-name>] [<template>] [<model>] [<description>]'}
```
Comment (kevgo) [inline src/cli.ls]: Yellow reads tough on white terminals.

**Comment by hugobho** [comment]
@kevgo All comments addressed

File: features/help.feature [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  So that I can do the things I intended without having to look this up somewhere else.
+
+
+  Scenario: the user enters 'exo add help'
+    When running "exo-add help" in the terminal
+    Then I see:
+    """
+    Usage: exo-add [<entity-name>]
+
+    Adds a service to this application...
```
Comment (kevgo) [inline features/help.feature]: The "..." were meant as placeholders for more text. Here is the text to use:

```
Ugase: exo-add [<service-name>] [<template>] [<model>] [<description>]

Adds a new service to the current application. 
This command must be called in the root directory of the application.
```

File: features/steps/steps.ls [hunk 7]
```diff
@@ -37,6 +37,19 @@ module.exports = ->
       cb!
     async.each table.rows!, enter-input, done
 
+  @When /^running "([^"]*)" in the terminal$/, (command, done) ->
+    app-dir := path.join process.cwd!, 'tmp'
+    fs.empty-dir-sync app-dir
```
Comment (kevgo) [inline features/steps/steps.ls]: Don't empty the dir here

File: src/cli.ls [hunk 8]
```diff
@@ -42,13 +46,25 @@ inquirer.prompt(questions).then (answers) ->
...
+
+
 
 # Returns the data the user provided on the command line,
 # and a list of questions that the user has to be asked still.
 function parse-command-line command-line-args
   data = {}
   questions = []
-  [_, _, service-type, service-name, template-name, model-name, description] = command-line-args
+  [_, _, entity-name, service-name, template-name, model-name, description] = command-line-args
```
Comment (kevgo) [inline src/cli.ls]: My bad for confusing this. `service-type` was correct. Sorry.

Comment (hugobho) [inline src/cli.ls]: @kevgo What is `service-type`? As off right now, that variable isn't actually used anywhere.

Comment (kevgo) [inline src/cli.ls]: That's the old value of it. Please take a look at the diff. Your PR renames it to `entity-name`. I think the old name was more appropriate.

Comment (hugobho) [inline src/cli.ls]: Right, but even then before I changed it to `entity-name`, I don't think the variable `service-type` was being used anywhere. @kevgo

**Comment by trushton** [comment]
UPDATE: This PR addresses [Originate/exosphere-sdk#148](https://github.com/Originate/exosphere-sdk/issues/148)

File: src/cli.ls [hunk 9]
```diff
@@ -42,13 +46,25 @@ inquirer.prompt(questions).then (answers) ->
 function service-names
   fs.readdir-sync path.join(templates-path, 'add-service')
 
+function help
+  console.log "\nUsage: " + cyan 'exo-add' + blue " [<entity-name>]\n"
```
Comment (kevgo) [inline src/cli.ls]: Use string interpolation here

File: src/cli.ls [hunk 10]
```diff
@@ -42,13 +46,25 @@ inquirer.prompt(questions).then (answers) ->
...
+  help-message =
+    """
+    Adds a service to this application...
+
+    options: #{blue '[<service-name>] [<template>] [<model>] [<description>]'}
+        Adds an Exoservice within application directory.
+    """
+  console.log help-message
```
Comment (kevgo) [inline src/cli.ls]: Make this whole method run only a single `console.log` message that outputs a single string. Use multi-line strings in LiveScript to get newlines (strings surrounded by `"""`).

**Comment by trushton** [comment]
@kevgo All comments addressed, I know these changes were approved a couple of months ago, just wanted to make sure its still good to go

File: features/steps/steps.ls [hunk 11]
```diff
@@ -38,6 +38,20 @@ module.exports = ->
     async.each table.rows!, enter-input, done
 
 
+  @When /^running "([^"]*)" in the terminal$/, (command, done) ->
+    app-dir := path.join process.cwd!, 'tmp'
+    #fs.empty-dir-sync app-dir
```
Comment (kevgo) [inline features/steps/steps.ls]: Delete this line

**Pull request closed without merging**